### PR TITLE
feat: add tracking to save article

### PR DESCRIPTION
**Pull request by Ivaylo1987**
Small PR for adding tacking to article save event.

### CODE REVIEW

File: packages/edition-slices/src/tiles/shared/save-article-tracking-events.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { withTrackEvents } from "@times-components/tracking";
+
+export default Component =>
```
[the remark this answers is not in the document]
Comment (nathggns) [reply packages/edition-slices/src/tiles/shared/save-article-tracking-events.js]: I think you'd probably call this `withArticleSaveTracking`

[the remark this answers is not in the document]
Comment (Ivaylo1987) [reply packages/edition-slices/src/tiles/shared/save-article-tracking-events.js]: 👍

**Comment by nathggns** [comment]
Reviewing now :)

File: packages/section/__tests__/shared-tracking.js [hunk 2]
```diff
@@ -71,6 +74,31 @@ export default () => {
     expect(onArticlePress.mock.calls).toMatchSnapshot("onArticlePress");
   });
 
+  it("Save/Unsave article click tracking", () => {
```
Comment (nathggns) [inline packages/section/__tests__/shared-tracking.js]: Do you need a test for the second time to make sure `isSaved` is false in the tracking?

File: packages/section/__tests__/shared-tracking.js [hunk 3]
```diff
@@ -71,6 +74,31 @@ export default () => {
...
+          section={edition.sections[0]}
+          stream={stream}
+        />
+      </SectionContext.Provider>
+    );
+
+    const [starButton] = testInstance.root.findAllByType(StarButton);
+    starButton.props.onPress();
+
+    const [[call]] = stream.mock.calls;
```
Comment (nathggns) [inline packages/section/__tests__/shared-tracking.js]: This is a difficult syntax to read with not much benefit – I'd prefer just the classic:

```js
const call = stream.mock.calls[0][0];
```

Comment (Ivaylo1987) [reply packages/section/__tests__/shared-tracking.js]: So, you dont want us to be cool ;)

File: packages/section/__tests__/shared-tracking.js (hunk 2 above)
Comment (Ivaylo1987) [reply packages/section/__tests__/shared-tracking.js]: We can try this 👍

File: packages/section/__tests__/shared-tracking.js (hunk 3 above)
Comment (nathggns) [reply packages/section/__tests__/shared-tracking.js]: I think there are benefits when you're destructing multiple elements from an array, or during function argument destructing, but in this case its just harder to read.

Comment (Ivaylo1987) [reply packages/section/__tests__/shared-tracking.js]: Sure, I agree will fix it :)

File: packages/section/__tests__/shared-tracking.js (hunk 2 above)
Comment (Ivaylo1987) [reply packages/section/__tests__/shared-tracking.js]: Done!

File: packages/section/__tests__/shared-tracking.js (hunk 3 above)
Comment (Ivaylo1987) [reply packages/section/__tests__/shared-tracking.js]: done!

**Pull request merged**